### ISSUE TITLE
libretro stella core vs stella2014 core 

### ISSUE THREAD

**Issue by alijani1** [libretro/stella2014-libretro#83]
Anyone have an idea on how we can get some of the fixes made here on stella2014 into the standard llibretro-stella core:  [https://github.com/libretro/stella](https://github.com/libretro/stella)

emuelec images do not have stella2014 cores installed by default and on many of the systems they are a paid to install manually.

any thoughts?

**Comment by alijani1** [comment]
Anyone was able to get this working with batocera or emuelec ?

**Comment by hizzlekizzle** [comment]
What are some of the things you would like to see? I was under the impression the up-to-date core was better in every way aside from speed.

**Comment by alijani1** [comment]
@hizzlekizzle, what I am specifically interested in is the fixed that was done under [issue 78](libretro/stella2014-libretro#78)

you can see the video on how great [the fix is in lr-Stella2014 here](https://www.youtube.com/watch?v=BYivtbp9cHs)

**Comment by keithbowes** [comment]
I myself have wondered why this core continues to be developed instead of putting new functionality in the upstream core and backporting relevant changes here (I know it's difficult to backport, due to the libretro frontends of two cores being so different; I implemented input bitmasks in upstream Stella, and it was far from just copying and pasting from Stella 2014).  I've seen issues in upstream Stella about incorporating things from Stella 2014, but I guess no one has gotten around to it yet.

BTW, what you want is https://github.com/stella-emu/stella.  https://github.com/libretro/stella is miles behind.

**Comment by thrust26** [comment]
The Stella team would **really** like to see that the mainline Stella would be better supported by the Libretro team. We are quite frequently getting requests regarding Libretro and the latest Stella and usually do not have the resources (or knowledge) to support them.

We would **much** prefer that development here switches to mainline Stella first. With optional back porting to Stella2014. Instead of vice versa as it currently is.

**Comment by hizzlekizzle** [comment]
I fully agree. Unfortunately, many of the downstream projects use the older fork for potato hardware (SBCs, embedded, etc.). Additionally, I think one of the primary hangups is just that we aren't even aware of libretro-focused issues unless they're reported here.I think one of the primary hangups is just that we aren't even aware of libretro-focused issues unless they're reported here.

I wonder if there's a good workflow for getting our contributors' eyes on the upstream issues..? What about a libretro repo that holds nothing but a submodule pointing to the upstream repo, and you guys could direct users to report libretro-specific issues there? and if/when we fix them, they would necessarily have to be submitted upstream?

**Comment by thrust26** [comment]
Adding paddle support for Libretro (like requested above) and other controllers shouldn't be a major problem for anyone know Libretro. Of course the Stella team would provide any necessary support here.

**Comment by alijani1** [comment]
> The Stella team would **really** like to see that the mainline Stella would be better supported by the Libretro team. We are quite frequently getting requests regarding Libretro and the latest Stella and usually do not have the resources (or knowledge) to support them.
> 
> We would **much** prefer that development here switches to mainline Stella first. With optional back porting to Stella2014. Instead of vice versa as it currently is.

First, am I correct that the main libreto stella core is here? [https://github.com/libretro/stella](https://github.com/libretro/stella) 

It looks like it's not been updated for over 3 years :(

@thrust26 , it looks like @hisslekizzle is willing to have contributors assist and was suggesting a possible workflow. any thoughts on his suggestion?

**Comment by hizzlekizzle** [comment]
AFAIK, we build and distribute [from the upstream repo](https://github.com/stella-emu/stella/blob/master/.gitlab-ci.yml), so anyone wanting to work on the libretro integration can fork the upstream repo, make changes, we can discuss/build/test those changes from the fork and then you can PR them upstream.

**Comment by thrust26** [comment]
Sounds good to me.

**Comment by theflyingape** [comment]
Latest RetroArch git source now fails to render Stella screen while all other cores work perfectly fine.
